command/server/tls_util.go

```diff
@@ -61,7 +61,7 @@ func GenerateCert(caCertTemplate *x509.Certificate, caSigner crypto.Signer) (str
 		NotAfter:       time.Now().Add(365 * 24 * time.Hour),
 		NotBefore:      time.Now().Add(-1 * time.Minute),
 		IPAddresses:    []net.IP{net.ParseIP("127.0.0.1")},
-		DNSNames:       []string{"localhost", "localhost4", "localhost6", "localhost.localdomain"},
+		DNSNames:       []string{hostname, "localhost", "localhost4", "localhost6", "localhost.localdomain"},
```

Comment:
It looks like if `hostname` is empty above, it's set to `localhost`. Should we instead check the value of hostname after we make the cert template and append it to the slice if its not localhost? Else we'll have duplicate localhost entries.